doc/credentials.md

```diff
@@ -0,0 +1,223 @@
+    "secretData":[
+        {
+            "key":"username",
+            "value":"<username>"
+        },
+        {
+            "key":"password",
+            "value":"<password>"
+        },
+        {
+            "key":"mode",
+            "value":"<mode>"
+        }
+    ]
+}'
+```
+
+## Mapping Credentials to Devices
+> **Note:** Credential mappings can be set via [utility scripts](./utility-scripts.md) after the service is running
+
+The device service supports three types of credential mapping. All three types can be used
+in conjunction with each other.
+
+- `1 to All` - All devices are given the default credentials based on `DefaultSecretPath`
+- `1 to Many` - In the `CredentialsMap`, one secret path can be assigned multiple MAC addresses
+- `1 to 1` - In the `CredentialsMap`, assign each secret path 1 MAC Address
+
+### Manual Configuration
+> **Note:** Any key present in `AppCustom.CredentialsMap` must also exist in the secret store!
+```toml
+# AppCustom.CredentialsMap is a map of SecretPath -> Comma separated list of mac addresses.
+# Every SecretPath used here must also exist as a valid secret in the Secret Store.
+#
+# Note: Anything not defined here will be assigned the default credentials configured via `DefaultSecretPath`.
+#
+# Example: (Single mapping for 1 mac address to 1 credential)
+#   credentials001 = "aa:bb:cc:dd:ee:ff"
+#
+# Example: (Multi mapping for 3 mac address to 1 shared credentials)
+#   credentials002 = "11:22:33:44:55:66,ff:ee:dd:cc:bb:aa,ab:12:12:34:34:56:56"
+#
+# The special group 'NoAuth' defines mac addresses of cameras where no authentication is needed.
+# The 'NoAuth' group does not exist in the SecretStore. It is not required to place cameras in here,
+# however it avoids sending the default credentials to cameras which do not need it.
+#
+# IMPORTANT: A MAC Address may only exist in one credential group. If a MAC address is defined in more
+# than one group, it is random which group the MAC will end up in! If you wish to change the group a MAC
+# address belongs to, first remove it from its existing group, and then add it to the new one.
+[AppCustom.CredentialsMap]
+NoAuth = "44:33:22:11:33:fd"
+credentials001 = "aa:bb:cc:dd:ee:ff"
+credentials002 = "11:22:33:44:55:66,ff:ee:dd:cc:bb:aa,ab:12:12:34:34:56:56"
+```
+
+## Credential Lookup
+Here is an in-depth look at the logic behind mapping `Credentials` to Devices.
+
+### During Discovery
+```mermaid
+%% Note: The node and edge definitions are split up to make it easier to adjust the
+%% links between the various nodes.
+graph TD;   
+    %% -------- Node Definitions -------- %%
+    discover[Discovered Device]
+    default[Use Default Credentials]
+    refmac{Does EndpointRef<br/>contain<br/>MAC Address?}
+    noauth{MAC Belongs<br/>to NoAuth group?}
+    none[Set AuthMode to 'none']
+    apply[Apply Credentials]
+    instore{Credentials exist<br/>in SecretStore?}
+    info[Get Device Information]
+    success{Success?}
```

Comment:
Success seems to be dangling in space.

Reply:
Maybe GitHub renderer... ?

Reply:
ah i originally had success as a decision and removed it in favor of just naming the arrows. guess i missed a spot.